flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -545,23 +545,29 @@ private void stopTaskExecutorServices() throws Exception {
 
     @Override
     public CompletableFuture<TaskThreadInfoResponse> requestThreadInfoSamples(
-            final ExecutionAttemptID taskExecutionAttemptId,
+            final Set<ExecutionAttemptID> taskExecutionAttemptIds,
```

Comment:
I was not sure if there could be an edge case of getting the same ExecutionAttemptID multiple times and wanted to avoid "double" representation in the FlameGraphs. But I agree, this probably should never happen.